### PR TITLE
Add NetworkListener class.

### PR DESCRIPTION
**Pull request by wurmmi**
Continue to implement class diagram structure.

### CODE REVIEW

File: src/NetworkListener.h [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+//#include "LoggingHandler.h"
+//#include "NetworkAPI.h"
+
+/**
+ * @brief Type for multiple queues
+ */
+using Queues = Queue[2];
+
+class NetworkListener {
+ public:
```
Comment (skaupper) [inline src/NetworkListener.h]: is this 1 space indent intended?

File: src/NetworkListener.h [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  virtual void start(std::string configFilePath) = 0;
```
Comment (skaupper) [inline src/NetworkListener.h]: i dont think start is needed here?

File: src/NetworkListener.h (hunk 1 above)
Comment (wurmmi) [reply src/NetworkListener.h]: yes, this is done by clang-format

File: CMakeLists.txt [hunk 3]
```diff
@@ -37,7 +37,16 @@ set(APP_SOURCES         src/JukeBox.cpp
...
-set(APP_HEADER          )
+
+set(APP_HEADER          src/JukeBox.h
+                        src/Track.h
+                        src/Queue.h
+                        src/LoggingHandler.h
+                        src/ConfigHandler.h
+                        src/NetworkListener.h
+                        src/Result.h
+                        src/GlobalTypes.h)
```
Comment (wurmmi) [inline CMakeLists.txt]: this needs a follow-up PR to update documentation through `make doc`...

**Comment by wurmmi** [comment]
Ready to merge from my side. 

Again - this is an early merge, in order to unblock other team members.

**Pull request merged**